File: app/src/main/java/com/jcons/pdfconverter/pdfconverter/List.java
package com.jcons.pdfconverter.pdfconverter;


import java.util.ArrayList;

public abstract class List extends Base {

    protected ArrayList<String> mList  = new ArrayList<String>();

    protected String renderList() {
        StringBuilder sb = new StringBuilder();
        for (String s : mList) {
            sb.append(s);
        }
        return sb.toString();
    }

    @Override
    public void clear() {
        mList.clear();
    }
}

File: app/src/main/java/com/jcons/pdfconverter/pdfconverter/List.kt
package com.jcons.pdfconverter.pdfconverter

import java.util.*


abstract class List : Base() {
    protected var mList = ArrayList<String>()
    protected fun renderList(): String {
        val sb = StringBuilder()
        for (s in mList) {
            sb.append(s)
        }
        return sb.toString()
    }

    override fun clear() {
        mList.clear()
    }
}
